drivers/sensor/ntc_thermistor/ntc_thermistor.c

```diff
@@ -0,0 +1,168 @@
+		data->sample_val = val_mv;
+	}
+
+	k_mutex_unlock(&data->mutex);
+
+	return res;
+}
+
+static int ntc_thermistor_channel_get(const struct device *dev, enum sensor_channel chan,
+				      struct sensor_value *val)
+{
+	struct ntc_thermistor_data *data = dev->data;
+	const struct ntc_thermistor_config *cfg = dev->config;
+	uint32_t ohm, max_adc;
+	int32_t temp;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		max_adc = (1 << (cfg->adc_channel.resolution - 1)) - 1;
+		ohm = ntc_get_ohm_of_thermistor(&cfg->ntc_cfg, max_adc, data->raw);
+		temp = ntc_get_temp_mc(&cfg->ntc_cfg.type, ohm);
+		val->val1 = temp / 1000;
+		val->val2 = (temp % 1000) * 1000;
+		LOG_INF("ntc temp says %u", val->val1);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static const struct sensor_driver_api ntc_thermistor_driver_api = {
+	.sample_fetch = ntc_thermistor_sample_fetch,
+	.channel_get = ntc_thermistor_channel_get,
+};
+
+static int ntc_thermistor_init(const struct device *dev)
+{
+	const struct ntc_thermistor_config *cfg = dev->config;
+	int err;
+
+	if (!device_is_ready(cfg->adc_channel.dev)) {
+		LOG_ERR("ADC controller device is not ready\n");
+		return -ENODEV;
+	}
+
+	err = adc_channel_setup_dt(&cfg->adc_channel);
+	if (err < 0) {
+		LOG_ERR("Could not setup channel err(%d)\n", err);
+		return err;
+	}
+
+	return 0;
+}
+
+#define NTC_THERMISTOR_DEFINE(inst, id, comp_table)                                                \
+	BUILD_ASSERT(ARRAY_SIZE(comp_table) % 2 == 0, "Compensation table needs to be even size"); \
+                                                                                                   \
+	static int compare_ohm_##id##inst(const void *key, const void *element);                   \
+                                                                                                   \
+	static struct ntc_thermistor_data ntc_thermistor_driver_##id##inst;                        \
+                                                                                                   \
+	static const struct ntc_thermistor_config ntc_thermistor_cfg_##id##inst = {                \
+		.adc_channel = ADC_DT_SPEC_INST_GET(inst),                                         \
+		.ntc_cfg =                                                                         \
+			{                                                                          \
+				.r25_ohm = DT_INST_PROP(inst, r25_ohm),                            \
+				.pullup_uv = DT_INST_PROP(inst, pullup_uv),                        \
+				.pullup_ohm = DT_INST_PROP(inst, pullup_ohm),                      \
+				.pulldown_ohm = DT_INST_PROP(inst, pulldown_ohm),                  \
+				.connected_positive = DT_INST_PROP(inst, connected_positive),      \
+				.type = {                                                          \
+					.comp = (const struct ntc_compensation *)comp_table,       \
+					.n_comp = ARRAY_SIZE(comp_table),                          \
```

Comment:
@gmarull stumbled on this line when trying to rebase #57341 on top of this.

I think the array size is a bit misleading, since it refers to the `comp_table` array which has 2x members (two `uint32_t`s for each `struct ntc_compensation`) and not to the `comp` member. I assume that's also the reason for changes in the `bsearch` helper function (`element_idx == type->n_comp - 1` -> `element_idx == (type->n_comp / 2) - 1`). But there's still places where `n_comp` is not updated to account for this change (e.g. in `ntc_lookup_comp`).

Wouldn't it be cleaner to simply set `n_comp` to the real `comp_table` size, like so?
```c
.n_comp = ARRAY_SIZE(comp_table) / 2,
```